tests/mistake_scanner/mistake_scanner.sh

```diff
@@ -0,0 +1,15 @@
+# Search for `return S2N_ERR_TYPE`. Should be `return S2N_ERROR(S2N_ERR_TYPE);`
+
+grep --exclude-dir="tests" --exclude-dir="docs" -rE "return[ ]+S2N_ERR_TYPE" ../../
```

Comment:
We probably don't want to grep for the string literal "S2N_ERR_TYPE", but rather use a regex that matches all the different error types defined here: https://github.com/awslabs/s2n/blob/master/error/s2n_errno.h#L37

I think "S2N_ERR_[A-Z0-9_]+" should work.